src/os/bluestore/BlueFS.cc

```diff
@@ -3038,54 +3036,67 @@ void BlueFS::_consume_dirty(uint64_t seq)
   }
 }
 
-// Extends log if its free space is smaller then bluefs_min_log_runway.
-// Returns space available *BEFORE* adding new space. Signed for additional <0 detection.
-int64_t BlueFS::_maybe_extend_log()
-{
+int64_t BlueFS::_maybe_extend_log(uint64_t amount, bool ignore_compaction) {
+  uint64_t runway = log.writer->file->fnode.get_allocated() - log.writer->get_effective_write_pos();
+  // increasing the size of the log involves adding a OP_FILE_UPDATE_INC which its size will 
+  // increase with respect the number of extents. bluefs_min_log_runway should cover the max size 
+  // a log can get.
+  if (runway < cct->_conf->bluefs_min_log_runway) {
```

Comment:
@aclamk Finer tuned allocations could happen here if we dynamically compute the size of the op_update_file_inc
```suggestion
  if (runway < log.t.op_file_udpate_inc_size(log.writer->file->fnode) {
```
The problem with this is the increasing number of extents.

Reply:
We do not want to have fine-grained allocations for bluefs log.
The concept here is to preallocate large area, and slowly trickle small writes to it.

For example, we do not update size of bluefs log. You can notice that `_replay()` is completely ok to read outside of EOF from it.